pytorch_lightning/plugins/training_type/fully_sharded.py

```diff
@@ -178,6 +182,19 @@ def lightning_module_state_dict(self) -> Dict[str, Union[Any, Tensor]]:
         # state dict.
         return super().lightning_module_state_dict()
 
+    def serialized_restore_model_state(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
+        checkpoint = {}
+        log.info(f"FullyShardedDataParallel has {self.num_processes} processes. Serializing to avoid CPU OOMs.")
+        for current_worker in range(self.num_processes):
+            if self.local_rank == current_worker:
+                checkpoint = super().load_checkpoint_file(checkpoint_path)
+                self.lightning_module.on_load_checkpoint(checkpoint)
+                self.load_model_state_dict(checkpoint)
+                log.info(f"Rank {self.global_rank}: done loading model states from {checkpoint_path}.")
+                del checkpoint["state_dict"]
+            self.barrier()
```

Comment:
this is pretty smart :) 

one concern, now the responsibility of calling the hooks is shifted to the plugin. do we want to allow that?
@PyTorchLightning/core-contributors

Reply:
In this case, I don't see a problem in it. The only thing I am afraid of, is that we may have a precedence case here and that might lead to this pattern where we don't want it.

What do you think @SeanNaren @awaelchli @ananthsub @carmocca ?